### PR TITLE
Improve ft2font error reporting.

### PR DESCRIPTION
**Pull request by anntzer**
This prints out human-readable error messages for failures.

As an example, `FT2Font("setup.py")` now raises
```
RuntimeError: In FT2Font: Can not load face (unknown file format; error code 0x2)
```
(sure, this specific error was special-cased before, but that's not the case for e.g. https://github.com/matplotlib/matplotlib/issues/22212, which is the motivation for this PR.)

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Review by QuLogic — APPROVED** [review]
I swear I tried to do this once, but I can't find any stash of it.

File: src/ft2font.cpp [hunk 1]
```diff
@@ -43,9 +43,23 @@
 
 FT_Library _ft2Library;
 
+// FreeType error codes; loaded as per fterror.h (modified to use std::map).
```
Comment (timhoffm) [inline src/ft2font.cpp:46]: I don't see any `std::map`. Is this still correct?

**Review by timhoffm — APPROVED** [review]
Modulo clarifying the comment. Can self merge after that.

File: src/ft2font.cpp (hunk 1 above)
Comment (anntzer) [reply src/ft2font.cpp]: Oops, that was some old implementation, good catch, fixed.

**Review by timhoffm — APPROVED** [review]

**Pull request merged**